### PR TITLE
Add follower read case

### PR DESCRIPTION
**Pull request by hunterlxt**
### What problem does this PR solve? 
Add follower read case

- add follower read cmd and test
- add new template and workflow for follower read test
- add workflow to test follower read in bank, bank2, sqllogic
- support follower read enable config for sqllogic, bank, bank2
- randomly enable follower read for tipocket mysql db (to test tiflash)

### CODE REVIEW

**Comment by hunterlxt** [comment]
@zhangjinpeng1987 This the PR for all follower read automated test

File: cmd/sqllogic/main.go [hunk 1]
```diff
@@ -34,10 +34,13 @@ var (
 	testDir         = flag.String("d", "sqllogictest", "test case dir")
 	taskCount       = flag.Int("t", 10, "concurrency")
 	skipError       = flag.Bool("skip-error", false, "skip error for query test")
+	replicaRead     = flag.String("tidb-replica-read", "leader", "tidb_replica_read mode, support values: leader / follower / leader-and-follower, default value: leader.")
 )
 
 func main() {
 	flag.Parse()
+	sqllogictest.ReplicaRead = *replicaRead
```
Comment (mahjonp) [inline cmd/sqllogic/main.go:42]: Use sqllogictest.Config to pass parameters.

**Comment by zhouqiang-cl** [comment]
@hunterlxt Can we review this pr now?

**Comment by hunterlxt** [comment]
@zhouqiang-cl yes

**Comment by zhouqiang-cl** [comment]
@mahjonp PTAL

**Comment by hunterlxt** [comment]
@mahjonp PTAL

File: argo/workflow/bank-follower-read.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+metadata:
+  generateName: bank-replica-read-
```
Comment (mahjonp) [inline argo/workflow/bank-follower-read.yaml:2]: use tipocket prefix here.

File: argo/workflow/bank-follower-read.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  arguments:
+    parameters:
+      - name: ns
+        value: bank-replica-read
+      - name: purge
+        value: "true"
+      - name: image_version
+        value: nightly
+      - name: storage_class
+        value: standard
```
Comment (mahjonp) [inline argo/workflow/bank-follower-read.yaml:15]:
```suggestion
        value: gp2
```

File: cmd/follower-read/main.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+		Config: cfg,
+		ClientCreator: follower.ClientCreator{
+			Cfg: &follower.Config{
+				DBName:      *dbname,
+				Concurrency: *concurrency,
+			},
+		},
+		Provisioner: cluster.NewK8sProvisioner(),
+		NemesisGens: util.ParseNemesisGenerators(fixture.Context.Nemesis),
+		VerifySuit:  verify.Suit{},
```
Comment (mahjonp) [inline cmd/follower-read/main.go:44]: you can delete this line.

File: tests/sqllogictest/client.go [hunk 5]
```diff
@@ -141,7 +143,7 @@ func (c *sqllogicClient) Start(ctx context.Context, _ interface{}, clientNodes [
 	taskChan := make(chan string, c.TaskCount)
 	doneChan := make(chan struct{}, c.TaskCount)
 	resultChan := make(chan *result, c.TaskCount)
-	dbs := createDatabases(c.TaskCount, fmt.Sprintf("%s:%d", c.ip, c.port), "root", "")
+	dbs := createDatabases(c.TaskCount, fmt.Sprintf("%s:%d", c.ip, c.port), "root", "", c.ReplicaRead)
```
Comment (mahjonp) [inline tests/sqllogictest/client.go:146]: the last argument is password here, it's empty.

**Pull request merged**